lib/webengine.cpp

```diff
@@ -67,39 +101,44 @@ void WebEngine::initialize(const QString &profilePath)
     webEngine->addComponentManifest(SAILFISHOS_WEBVIEW_MOZILLA_COMPONENTS_PATH + QString("/chrome/EmbedLiteJSScripts.manifest"));
     webEngine->addComponentManifest(SAILFISHOS_WEBVIEW_MOZILLA_COMPONENTS_PATH + QString("/chrome/EmbedLiteOverrides.manifest"));
 
-    QTimer::singleShot(0, webEngine, SLOT(runEmbedding()));
+    if (runEmbedding) {
+        QTimer::singleShot(0, webEngine, SLOT(runEmbedding()));
+    }
 
     isInitialized = true;
 }
 
 /*!
-  \brief Returns the instance of the singleton WebEngine class.
+    \brief Returns the instance of the singleton WebEngine class.
 
-  Returns the instance of the singleton WebEngine class.
-  The returned instance may not be intialised.
+    The returned instance may not be intialised.
 
-  \sa initialize
+    \sa initialize
 */
 WebEngine *WebEngine::instance()
 {
     return webEngineInstance();
 }
 
 /*!
-  \brief Constructs a new WebEngine instance, with the specified QObject \a parent.
+    \internal
+    \brief Constructs a new WebEngine instance, with the specified QObject \a parent.
 
-  In general there should be no need to use the constructor. Call \l instance
-  to return the singleton instance instead.
+    In general there should be no need to use the constructor. Call \l instance
+    to return the singleton instance instead.
 
-  \sa instance
+    \sa instance
 */
 WebEngine::WebEngine(QObject *parent)
     : QMozContext(parent)
 {
 }
 
 /*!
-  \brief Destroys the WebEngine instance.
+    \internal
+    \brief Destroys the WebEngine instance.
+
+    WebEngine is a singleton, so should never be deleted.
```

Comment:
.. by the client.  It will be automatically deleted by the QML engine at application shutdown.

Reply:
Thanks; I've changed this.